cpp/include/cudf/detail/date_sequence.cuh

```diff
@@ -0,0 +1,66 @@
+#include <sys/syscall.h>
+#include <chrono>
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_view.hpp>
+#include <cudf/datetime.hpp>
+#include <cudf/detail/datetime_ops.cuh>
+#include <cudf/scalar/scalar.hpp>
+#include <cudf/scalar/scalar_device_view.cuh>
+#include <cudf/types.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/exec_policy.hpp>
+
+namespace cudf {
+namespace detail {
+
+template <typename Timestamp>
+CUDA_DEVICE_CALLABLE Timestamp add_dateoffset(
+  cudf::timestamp_scalar_device_view<Timestamp> const initial, std::size_t n, std::size_t months)
+{
+  datetime::detail::add_calendrical_months_functor_impl f{};
+  // just add `n` days:
+  return f(initial.value(), n * months);
+}
+
+struct date_sequence_functor {
```

Comment:
Since only one type is supported, I feel like you could move the FAIL to a higher level, and then have a non-type-dispatched / non-template implementation.